src/Temporalio/Worker/Tuning/CustomSlotSupplier.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Temporalio.Worker.Tuning
+{
+    /// <summary>
+    /// This class can be implemented to provide custom slot supplier behavior.
+    /// </summary>
+    /// <remarks>
+    /// WARNING: Custom slot suppliers are currently experimental.
+    /// </remarks>
+    public abstract class CustomSlotSupplier : SlotSupplier
```

Comment:
I wonder if you have to have a public, parameterless constructor here for accessibility outside the project. The tests package doesn't do a good job of testing this because we set `System.Runtime.CompilerServices.InternalsVisibleTo` to the test project so it can access some visible-to-testing only things (I forget what). May need just a quick confirmation this can be extended outside the project. Can just be some scratch code in `tests/Temporalio.SmokeTests`, doesn't have to be committed.

Reply:
It appears to work just fine without it. The bench project is easier to use since it doesn't require nupkg mumbojumbo but also doesn't do internals visible.